### PR TITLE
Zax 2.0 and File Commands

### PR DESCRIPTION
**Pull request by lord-max**
Numerous stability and performance updates to the core codebase, and introduced new functionality of extending the Zax “dead drop” style communications to include file exchange.

- Codebase upgraded to `Ruby 2.4.1` and `Rails 5.1.3`
- New set of [commands](https://github.com/vault12/zax/wiki/Zax-2.0-File-Commands) for device-to-device exchange of large files
- Dynamic throttling option: when on, the relay session handshake “proof of work” function will grow harder with increased server load
- Restart time window: optionally config time periods when supporting services are restarting, all workers will sleep during that window
- Improvements and optimizing for multi-worker/multi-threading access to Redis
- New logging details and easier to read color-coded logs
- Many performance improvements and bug fixes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,8 +174,9 @@ cd zax
 . tools/relay
 . tools/redis # new console window
 ```
+
 ## Demo
-To see Glow and Zax in action, check out the [Live Demo](https://zax_test.vault12.com). This is a test project included in Zax called [Zax-Dash](https://github.com/vault12/zax-dash).
+To see Glow and Zax in action, check out the [Live Demo](https://zax-test.vault12.com). This is a test project included in Zax called [Zax-Dash](https://github.com/vault12/zax-dash).
```
Comment (commagere) [inline README.md]: Did you want to change this url to z.vault12.com?

File: public/index.html [hunk 2]
```diff
@@ -18,7 +18,7 @@
       <!-- Use this section to provide information about your relay -->
       <div class="panel panel-primary info-pane" ng-hide="infoVisible">
         <div class="panel-heading">
-          <h3 class="panel-title">INFO: This is a <i>'Zax Test'</i> instance at <b><span id="hostblock">https://zax_test.vault12.com</span></b>
+          <h3 class="panel-title">INFO: This is a <i>Zax</i> instance at <b><span id="hostblock"></span></b>
```
Comment (commagere) [inline public/index.html]: I think this is missing the info for the server url - don't you want https://z.vault12.com to be here?

Comment (lord-max) [reply public/index.html]: That is correct: client script writes in `#hostblock` url index is served from

File: README.md (hunk 1 above)
Comment (lord-max) [reply README.md]: `zax-test` is live, we had to discard `zax_test` because of "let's encrypt" bizarre limit.

**Pull request merged**